bokeh/server/core/server_document.py

```diff
@@ -0,0 +1,32 @@
+'''
+
+'''
+from __future__ import absolute_import
+
+from tornado import gen
+from tornado.queues import Queue
+
+class ServerDocument(object):
+
+    def __init__(self, docid):
+        self._docid = docid
+        self._queue = Queue
+        self._stopped = False
+
+    def stop_worker(self):
+        self._stopped = True
+
+    @gen.coroutine
+    def workon(self, item):
+        yield self._queue.put(item)
```

Comment:
For instance, a computationally expensive callback gets queued here, that eventually updates the data source for a plot. That causes another chunk of work to get queues on the main IOLoop that will either push updates to the connected client, or send a message that there are updates available to pull. Does that seem reasonable? I suppose the `yield` here could return that secondary work and queue it on the other queue here, though...